### PR TITLE
Fix #4829: render non-grouped knobs in the ALL tab

### LINKED ISSUE #4829: When group specified for any knob, others without group are not displaying
When set group for knob in story other knobs, that dont have group, are not displaying in knobs panel

**To Reproduce**
1. Add story with two knobs (`text` for example)
2. Set group to one of them
3. Open storybook
4. Knob without group is not in knob panel 

**Expected behavior**
Group 'All' should contain the knob without group

**Screenshots**
image

**Code snippets**
```
<Block variation={{
    endpoint: text('frame 1: endpoint', 'http://www.google.com', 'test'),
    height: text('frame 1: height', '1650px')
}}
```

**System:**
 - OS: Windows10
 - Device: Laptop Dell G3
 - Browser: Chrome
 - Framework: react
 - Addons: 
    * @storybook/addon-actions: "4.0.6",
    * @storybook/addon-comments: "3.2.19",
    * @storybook/addon-console: "1.1.0",
    * @storybook/addon-info: "4.0.6",
    * @storybook/addon-knobs: "4.0.6",
    * @storybook/addon-links: "4.0.6",
    * @storybook/addon-notes: "4.0.6",
    * @storybook/addon-options: "4.0.6",
    * @storybook/addon-storyshots: "4.0.6",
    * @storybook/addon-storyshots-puppeteer: "4.0.6",
    * @storybook/addon-viewport: "4.0.6",
 - Version: 4.0.6

### PR DESCRIPTION
**Pull request by Jessidhia**
Issue: #4829 

## What I did

When there are knobs both with and without a `groupId`, the knobs without a `groupId` were not getting rendered at all.

I made them be rendered as part of the content of the (normally contentless) `ALL` tab.

## How to test

- Is this testable with Jest or Chromatic screenshots?
  - I don't know about screenshots but I wrote Jest tests.
- Does this need an update to the documentation?
  - No, this fixes something that is documented to work but doesn't

---

PS: Oops, I only saw the notice about which branch to use as base after I wrote this. I need this to be in `master` FWIW. What do?

### CODE REVIEW

**Comment by ndelangen** [comment]
I'll rebase this

File: addons/knobs/src/components/__tests__/Panel.js [hunk 1]
```diff
@@ -144,4 +146,124 @@ describe('Panel', () => {
...
+              defaultValue: 'test2',
+              used: true,
+              groupId: 'bar',
+            },
+          },
+        });
+
+        const titles = wrapper
+          .find(TabsState)
+          // TabsState will replace the <div/> that Panel actually makes with a <Tab/>
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply addons/knobs/src/components/__tests__/Panel.js]: Would be happy to chat about how to improve this, Are you on the Storybook discord by any chance?

**Comment by ndelangen** [comment]
OK it's rebased, and I took the liberty of applying the changes for @igor-dv review comments.

@Kovensky we're migrating to typescript, and we're discussing how the future version of addon-knobs is going to look like / function on discord. I'd love your input on it, and help if you can. 🙇

**Review by ndelangen — APPROVED** [review]

File: addons/knobs/src/components/Panel.js [hunk 2]
```diff
@@ -20,10 +20,29 @@ const PanelWrapper = styled.div({
   width: '100%',
 });
 
+/** @typedef {{[knob: string]: { name: string; defaultValue: unknown; used: boolean; groupId?: string }}} Knobs */
```
[the remark this answers is not in the document]
Comment (Kovensky) [reply addons/knobs/src/components/Panel.js]: I thought they could be helpful when actually converting to typescript (these are all valid `// @ts-check` comments)

**Pull request merged**